components/list/list-item-drag-mixin.js

```diff
@@ -33,86 +95,299 @@ export const ListItemDragMixin = superclass => class extends superclass {
 		}
 	}
 
-	moveAfter(itemKey) {
+	moveItemAfterMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, false);
 	}
 
-	moveBefore(itemKey) {
+	moveItemBeforeMe(itemKey) {
 		if (itemKey === undefined) {
 			return;
 		}
 
-		this._dispatchDragEvent(itemKey);
+		this._moveItem(itemKey, this.key, true);
+	}
+
+	_dragAreaClick(e) {
+		this.shadowRoot.querySelector(`#${this._itemDragId}`).activateKeyboardMode();
+		e.preventDefault();
 	}
 
-	// TODO
-	_copy() {
+	_dragBottomEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		const dropSpots = dropSpotsFactory();
+		dropSpots.setDestination(this, move.below);
+		this._inBottomArea = true;
+	}
 
+	_dragEnd(e) {
+		const dropSpot = dropSpotsBlowUp();
+		this.beingDragged = false;
+		this.tilt = false;
+		if (dropSpot.shouldDrop(e.timeStamp)) {
+			this._moveItem(dropSpot.targetKey, dropSpot.destinationKey, dropSpot.moveBeforeDestination);
+		}
```

Comment:
Instead of moving the item during the drop event I move it here. This is because it gives that 1 second area around the drop areas for a better user experience.

I also, using the timestamps produced by the events for a consistent differences in time.